File: leet/0199-binary-tree-right-side-view/binary-tree-right-side-view.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution:
    def rightSideView(self, root: TreeNode) -> List[int]:
        if root is None: return []
        
        queue = [root]
        ret = []
        
        while len(queue) > 0:
            ret.append(queue[-1].val)
            
            tempQ = []
            for n in queue:
                if n.left is not None:
                    tempQ.append(n.left)
                if n.right is not None:
                    tempQ.append(n.right)
            
            queue = tempQ
            
        return ret
        

File: leet/0199-binary-tree-right-side-view/binary-tree-right-side-view.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val) {
 *     this.val = val;
 *     this.left = this.right = null;
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number[]}
 */
var rightSideView = function(root) {
    if (!root) return []
    
    let ret = []
    
    let temp = [root]
    
    while(temp.length > 0){
        ret.push(temp[temp.length - 1].val)
        
        let newT = []
        for (let n of temp){
            if (n.left) newT.push(n.left)
            if (n.right) newT.push(n.right)
        }
        temp = newT
    }
    
    return ret
};
